tests/components/vizio/test_config_flow.py

```diff
@@ -0,0 +1,294 @@
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_NAME: NAME,
+            CONF_HOST: HOST,
+            CONF_DEVICE_CLASS: DEVICE_CLASS_TV,
+            CONF_ACCESS_TOKEN: ACCESS_TOKEN,
+        },
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == NAME
+    assert result["data"][CONF_NAME] == NAME
+    assert result["data"][CONF_HOST] == HOST
+    assert result["data"][CONF_DEVICE_CLASS] == DEVICE_CLASS_TV
+    assert result["data"][CONF_ACCESS_TOKEN] == ACCESS_TOKEN
+    await hass.config_entries.async_unload(result["result"].entry_id)
+
+
+async def test_user_host_already_configured(
+    hass: HomeAssistantType, vizio_connect
+) -> None:
+    """Test host is already configured during user setup."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_SOUNDBAR_ENTRY,
+        options={CONF_VOLUME_STEP: VOLUME_STEP},
+    )
+    entry.add_to_hass(hass)
+    fail_entry = MOCK_SOUNDBAR_ENTRY.copy()
+    fail_entry[CONF_NAME] = "newtestname"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=fail_entry,
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_HOST: "host_exists"}
+
+
+async def test_user_name_already_configured(
+    hass: HomeAssistantType, vizio_connect
+) -> None:
+    """Test name is already configured during user setup."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_SOUNDBAR_ENTRY,
+        options={CONF_VOLUME_STEP: VOLUME_STEP},
+    )
+    entry.add_to_hass(hass)
+
+    fail_entry = MOCK_SOUNDBAR_ENTRY.copy()
+    fail_entry[CONF_HOST] = "0.0.0.0"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], fail_entry
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_NAME: "name_exists"}
+
+
+async def test_user_error_on_could_not_connect(
+    hass: HomeAssistantType, vizio_cant_connect
+) -> None:
+    """Test with could_not_connect during user_setup."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    with patch("pyvizio.vizio.VizioAsync.can_connect", return_value=False):
```

Comment:
We're already patching `validate_ha_config` on the VizioAsync instance. We don't need this patch here too.

Reply:
missed this when doing the cleanup, thanks